Финальный проект 4 спринта

alenaa-bva opened the pull request.

alenaa-bva committed:
Финальный проект 4 спринта

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,153 @@
-    # обязательно указывать префикс test_
-    # дальше идет название метода, который тестируем add_new_book_
-    # затем, что тестируем add_two_books - добавление двух книг
+    book_names = ['Тайна старого замка:ключ к древней тайне', 'Гордость и предубеждение и зомби', '?', 'N', ' ']
+    book_genres = ['Ужасы', 'Фантастика','Детективы', 'Комедии', 'Мультфильмы']
+
+    book_names_and_genres = [
+        ['Тайна старого замка:ключ к древней тайне', 'Ужасы'],
+        ['Гордость и предубеждение и зомби', 'Фантастика'],
+        ['?', 'Детективы'],
+        ['N', 'Комедии'],
+        [' ', 'Мультфильмы']
+    ]
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще, в тетсах и тестовых классах их держать не стоит

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,153 @@
+        if genre in ['Фантастика', 'Комедии', 'Мультфильмы']:
+            assert book in list_of_books_for_children
+        else:
+            assert book not in list_of_books_for_children
```
Нужно исправить: условий в тестах быть не может, тест всегда однозначен. Если это не вписывается в параметризацию, значит ее быть не должно и это просто несклько отдельных тестовых методов

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,153 @@
+        assert book_names[1] in list_of_favourite_books
+        assert book_names[3] in list_of_favourite_books
+
+
```
Нужно исправить: не проверен метод .get_books_genre

alenaa-bva committed:
Тестовые данные вынесены в файл data.py.

Добавлены тесты:
test_get_books_genre_one_book
test_get_books_for_children_appropriate_genres_in_the_list
test_get_books_for_children_inappropriate_genres_not_in_the_list

Удалены ненужные тесты:
test_get_books_genre_initial_state
test_get_list_of_favorites_books_initial_state
